data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -67,7 +74,11 @@
   private final Schema requiredSchema;
   private final Accessor<StructLike> posAccessor;
 
-  protected DeleteFilter(FileScanTask task, Schema tableSchema, Schema requestedSchema) {
+  private int readDeletesWorkerPoolSize;
+  private ExecutorService readDeletesService;
+
+  protected DeleteFilter(FileScanTask task, Schema tableSchema, Schema requestedSchema,
+                         Map<String, String> tableProperties) {
```

Comment:
This class should not take table properties. I think it should take an executor service instead.

Reply:
I think `DeleteFilter` should make sure the reader will read delete files in parallel by itself, if we config it, but not pass a executor service by the caller. So I create the read service internally, and read the config from system properties.